File: server/src/main/java/net/serlith/version/server/database/types/ServerSoftware.java
package net.serlith.version.server.database.types;

import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

@NoArgsConstructor
@AllArgsConstructor
@Table(name = "version_server_software")
public class ServerSoftware {

    @Id
    public Long id;

    @Column("name")
    public String name;

    @Column("display_name")
    public String displayName;

}

File: src/main/kotlin/net/serlith/version/database/types/ServerSoftware.kt
package net.serlith.version.database.types

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint

@Entity
@Table(
    name = "server_software",
    uniqueConstraints = [
        UniqueConstraint(columnNames = ["software", "version"])
    ]
)
class ServerSoftware {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @Column(name = "software")
    lateinit var software: String

    @Column(name = "version")
    lateinit var version: String

    @Column(name = "build")
    var build: Int = 0


    // Likely unneeded
    // @OneToMany(mappedBy = "versions", cascade = [CascadeType.ALL], orphanRemoval = true)
    // var versions: MutableList<ServerVersion> = mutableListOf()

}